fuzz/test-corpus.c

```diff
@@ -52,6 +177,11 @@ static void testfile(const char *pathname)
     if (buf != NULL) {
         s = fread(buf, 1, st.st_size, f);
         OPENSSL_assert(s == (size_t)st.st_size);
+#ifdef ERROR_INJECT
+        if (s > 0)
+            while (my_rand() % 3 <= 1)
+                buf[my_rand() % s] = (unsigned char)my_rand();
```

Comment:
Why do we do this? It seems this deliberately corrupts the input file into the fuzzer.

Reply:
Yes, that is also error injection.  We have a 33% chance of using the test vector unmodified.
And 22% of one byte error, 14% of two byte errors, and so on.

Reply:
This seems quite counter-intuitive to me. We are basing the error injection approach on the various corpora files because we hope that the complete set hits as many paths through the codebase as possible. So we increase the likelihood of hitting a codepath with a memory allocation which we can then cause to fail or not.

By deliberately corrupting the corpora files its seems we are reducing the effectiveness of the approach. A corruption in the files seems to me more likely to cause an early bail-out which means we don't then hit the codepaths that the files would otherwise reach. So these corruptions seem like they would reduce the number of codepaths that we are going to hit in total in any one run through of the complete set of files.

In other words, this corruption seems more likely to decrease the chances of finding memory errors than increase them.